processmanager/process_impl.go

```diff
@@ -12,57 +12,96 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package process
+package processmanager
 
 import (
+	"io"
 	"os"
+	"os/exec"
 	"strings"
 	"syscall"
 	"time"
 
-	"github.com/ligato/cn-infra/process/status"
+	"github.com/ligato/cn-infra/processmanager/status"
 	"github.com/pkg/errors"
 )
 
 // Marked defines that the process should be always restarted
 const infiniteRestarts = -1
 
-func (p *Process) startProcess() (*os.Process, error) {
-	wd, err := os.Getwd()
+func (p *Process) startProcess() (cmd *exec.Cmd, err error) {
+	// set directory
+	cmd = &exec.Cmd{Path: p.cmd}
+	cmd.Dir, err = os.Getwd()
 	if err != nil {
 		return nil, errors.Errorf("failed to get rooted path name for: %v", err)
 	}
-	var attr = os.ProcAttr{
-		Dir:   wd,
-		Env:   os.Environ(),
-		Files: []*os.File{os.Stdin, nil, nil},
-	}
-	// Syscall if process should be detached from parent
-	if p.options != nil && p.options.detach {
-		attr.Sys = &syscall.SysProcAttr{
-			Setpgid: true,
-			Pgid:    0,
+
+	// if options are set, adjust command attributes, otherwise set last required fields to prepare the command
+	if p.options == nil {
```

Comment:
It seems that these if cases have redundant code. You could introduce variable `defaultOptions`, containing these options from first if case. Then the first case can be removed and default options will be set by the other one.